src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt

```diff
@@ -312,7 +312,7 @@ function create_and_run_classpath_jar() {
       path="file:${path}" # e.g. "file:/usr/local/foo.jar"
     fi
 
-    path=$(sed "s/ /%20/g" <<< "${path}")
+    path=$(sed "s/ /%20/g; s/%/%25/g" <<< "${path}")
```

Comment:
I'm not expert in sed. Do you care to explain how '%' and ' ' are suppose to be handled?
For example from space and percentage ' %', is the desired result '%2520%25' or '%20%25'. On first sight it seems that you'll get the first one. I've seen cases in the past where this is actually desired, but they were rare.

Is is possible by any chance to use urlencode and urldecode functions?

Reply:
As stated in the other comment, this is a bug and I'd prefer the second example rather than the first.

I'd love to use `urlencode` but it isn't available on `macos` which presents a problem. If you know of an alternative tool I'd be happy to use that instead since rolling our own url encoder, besides being a waste of time, would be error-prone.